src/controller/scan/base_controller.go

```diff
@@ -233,78 +266,116 @@ func (bc *basicController) Scan(ctx context.Context, artifact *ar.Artifact, opti
 	return nil
 }
 
-func (bc *basicController) makeReportPlaceholder(ctx context.Context, r *scanner.Registration, art *ar.Artifact, options ...Option) (string, []string, error) {
-	trackID, err := bc.uuid()
+func (bc *basicController) ScanAll(ctx context.Context, trigger string, async bool) (int64, error) {
+	extraAttrs := map[string]interface{}{}
+	executionID, err := bc.execMgr.Create(ctx, job.ImageScanAllJob, 0, trigger, extraAttrs)
 	if err != nil {
-		return "", nil, errors.Wrap(err, "scan controller: scan")
+		return 0, err
 	}
 
-	// Parse options
-	ops, err := parseOptions(options...)
-	if err != nil {
-		return "", nil, errors.Wrap(err, "scan controller: scan")
+	if async {
+		go bc.startScanAll(bc.makeCtx(), executionID, async)
+	} else {
```

Comment:
is it better putting the checking for `async` inside the function `startScanAll`?

Reply:
`async` was removed from the parameters of the `startScanAll`